app/src/organisms/OnDeviceDisplay/RobotDashboard/hooks/useMissingHardwareChipText.ts

```diff
@@ -0,0 +1,48 @@
+import { useTranslation } from 'react-i18next'
+import type { ProtocolHardware } from '../../../../pages/Protocols/hooks'
+
+export function useMissingHardwareChipText(
+  missingProtocolHardware: ProtocolHardware[]
+): string {
+  const { t, i18n } = useTranslation('device_details')
+  const missingProtocolHardwareType = missingProtocolHardware.map(
+    hardware => hardware.hardwareType
+  )
+  // Note(kj:04/13/2023) This component only check the type and count the number
+  // If we need to display any specific information, we will need to use filter
+  const countMissingHardwareType = (hwType: 'pipette' | 'module'): number => {
+    return missingProtocolHardwareType.reduce((acc, hardwareType) => {
+      if (hardwareType === hwType) {
+        return acc + 1
+      }
+      return acc
+    }, 0)
+  }
```

Comment:
Changed my mind 🙃. I think probably the following would be easy to understand. I think for performance `reduce` would be better than filter but this is just checking recent run protocols (max number is 8) so readability would be more important for us. Of course, this is optional 😄  

```suggestion
const countMissingHardwareType = (hwType: 'pipette' | 'module'): number => {
  return missingProtocolHardwareType.filter(hardwareType => hardwareType === hwType).length
}
```